### PR TITLE
c_softmax_with_cross_entropy_op supports multiple label values

### PR DESCRIPTION
**Pull request by tianhaodongbd**
### PR Category

Distributed Strategy

### PR Types

New features

### Description

ParallelCrossEntropy 支持多标签分类
c_softmax_with_cross_entropy_op 的label参数支持shape=[B, S, C]，其中1 <= C <= vocab_size
Pcard-87604

### CODE REVIEW

File: paddle/fluid/operators/collective/c_softmax_with_cross_entropy_op.cu [hunk 1]
```diff
@@ -78,15 +81,72 @@ __global__ void CaculateLoss(T* loss,
...
+                             predict_logits[i * C + j]) *
+                            prob);
+      ignore_num += ignore_index == real_label ? 1 : 0;
+    }
+    loss[i] = ignore_num > 0 ? static_cast<T>(0) : tmp_loss;
+  }
+}
+
+template <typename T, typename IndexT>
+__global__ void CaculateLogitsGrad(T* logits_grad,
```
Comment (GuoxiaWang) [inline paddle/fluid/operators/collective/c_softmax_with_cross_entropy_op.cu]: 这里`CaculateLogitsGrad` 改成 `CaculateSoftLogitsGrad` 吧

File: paddle/fluid/operators/collective/c_softmax_with_cross_entropy_op.cu [hunk 2]
```diff
@@ -503,35 +573,96 @@ class CSoftmaxWithCrossEntropyGradCUDAKernel : public framework::OpKernel<T> {
...
     phi::DenseTensor logit_grad_2d;
     logit_grad_2d.ShareDataWith(*logit_grad).Resize({N, D});
 
     int64_t blocks = NumBlocks(N * D);
+    int64_t blocks_cal = NumBlocks(N);
     int threads = kNumCUDAThreads;
     const auto& label_type = framework::TransToProtoVarType(labels->dtype());
     const int64_t start_index = rank * D;
     const int64_t end_index = start_index + D;
+    phi::DenseTensor is_ignore;
```
Comment (GuoxiaWang) [inline paddle/fluid/operators/collective/c_softmax_with_cross_entropy_op.cu]: `phi::DenseTensor is_ignore;` 放到下面 `if (C > 1) {` 里面去，把作用缩小。

**Review by GuoxiaWang — CHANGES_REQUESTED** [review]

File: paddle/fluid/operators/collective/c_softmax_with_cross_entropy_op.cu (hunk 1 above)
Comment (tianhaodongbd) [reply paddle/fluid/operators/collective/c_softmax_with_cross_entropy_op.cu]: done

File: paddle/fluid/operators/collective/c_softmax_with_cross_entropy_op.cu (hunk 2 above)
Comment (tianhaodongbd) [reply paddle/fluid/operators/collective/c_softmax_with_cross_entropy_op.cu]: done

**Review by luotao1 — APPROVED** [review]
LGTM for Paddle_enforce

**Review by GuoxiaWang — APPROVED** [review]
